lib/Sema/ConstraintSystem.cpp

```diff
@@ -4226,6 +4222,15 @@ bool constraints::isPatternMatchingOperator(Expr *expr) {
   return isOperator(expr, "~=");
 }
 
+bool constraints::isStandardComparisonOperator(Expr *expr) {
+  if (auto opName = getOperatorName(expr)) {
+    return opName->is("==") || opName->is("!=") || opName->is("===") ||
+           opName->is("!==") || opName->is("<") || opName->is(">") ||
+           opName->is("<=") || opName->is(">=");
```

Comment:
It would’ve been nice if we used the identifiers from `ASTContext` (like `Id_LessThanOperator`) instead of hardcoded literals, but some of the operators are missing from the [list](https://github.com/apple/swift/blob/6aeca31368cffdf9fbcd3ed3b87a81bd36f13938/include/swift/AST/KnownIdentifiers.def#L150) :/

Reply:
I'll refactor that separately, that's just logic I have extracted from `diagnoseAmbiguityWithFixes` :)